### ISSUE TITLE
SPM Dependencies are included in code coverage when using `tuist test`

### ISSUE THREAD

**Issue by wattson12** [tuist/tuist#4159]
**Describe the bug**
When running tests using `tuist test` targets which are included as a result of SPM dependencies added to `Tuist/Dependencies.swift` are included in the list of targets to include in code coverage reports. Since these targets don't have their tests run the coverage will be 0 (unless executed by your own frameworks / tests)

**To Reproduce**

1. Unzip the attached project which has a single target (`A`) and tests (`ATests`). `A` has a dependency on the external SPM library [ComposableArchitecture](https://github.com/pointfreeco/swift-composable-architecture)
2. Fetch dependencies: `tuist dependencies fetch`
3. Run tests and generate result file: `tuist test --result-bundle-path demo.xcresult`
4. Open `demo.xcresult` and select the code coverage report, you will see frameworks from the `ComposableArchitecture` package, e.g. `CasePaths.framework`

**Expected behavior**

For the demo app above the only framework which should be included in the coverage report is `A.framework` 

**Screenshots**

<img width="930" alt="Screenshot on 2022-02-16 at 13-12-10" src="https://user-images.githubusercontent.com/1217873/154271656-0a9a0422-1a01-4d28-b18e-4ad5546e6072.png">

**Desktop (please complete the following information):**

- OS: `macOS 12.0.1`
- Version: tuist version `2.7.2`

**Additional context**

Demo project [tuist_dependency_coverage.zip](https://github.com/tuist/tuist/files/8080106/tuist_dependency_coverage.zip)

**Comment by danyf90** [comment]
I think it might be fixed in 3.0, could you please try with that from main? 🙏

**Comment by wattson12** [comment]
> I think it might be fixed in 3.0, could you please try with that from main? 🙏

I don't have much more time to test, I've tried to update the config to include this in the workspace 

```
generationOptions: .options(autogeneratedWorkspaceSchemes: .enabled(codeCoverageMode: .relevant))
```
and this in the project 
```
options: .options(automaticSchemesOptions: .enabled(codeCoverageEnabled: true)),
```
but the generated report doesn't seem to include coverage, I might be missing something though (I haven't kept up to date with the 3.0 changes)

<img width="280" alt="Screenshot on 2022-02-16 at 14-45-27" src="https://user-images.githubusercontent.com/1217873/154288934-74ef0e13-50d7-4427-9862-cd731e510509.png">

**Comment by connor-ricks** [comment]
This still seems to be an issue.
